classes/local/store/object_file_system.php

```diff
@@ -737,7 +737,26 @@ protected function copy_from_external_to_local($contenthash) {
      */
     public function redirect_to_presigned_url($contenthash, $headers = array()) {
         try {
-            redirect($this->externalclient->generate_presigned_url($contenthash, $headers));
+            $signedurl = $this->externalclient->generate_presigned_url($contenthash, $headers);
+            if (headers_sent()) {
+                debugging('Headers already sent! This redirect may be incorrectly cached in browser');
```

Comment:
Can you just stuff this message with some useful context, mention objectfs and the url we are serving

Reply:
Done, I added the contenthash as well (probably not super useful but just in case).